client/state/data-layer/wpcom/sites/media/index.js

```diff
@@ -69,9 +69,13 @@ export const updateMediaSuccess = ( { siteId }, mediaItem ) => ( dispatch ) => {
 	dispatchFluxUpdateMediaItemSuccess( siteId, mediaItem );
 };
 
-export const updateMediaError = ( { siteId }, error ) => (/* dispatch, getState */) => {
-	dispatchFluxUpdateMediaItemError( siteId, error );
-};
+export const updateMediaError = ( { siteId, originalMediaItem } ) => [
+	receiveMedia( siteId, originalMediaItem ),
+	removeNotice( `update-media-notice-${ originalMediaItem.ID }` ),
```

Comment:
This shouldn't be necessary. Because we have the same `id` set, the previous notice should be automatically removed.

Reply:
It's true that this isn't necessary to prevent notice duplicates but there's a slight difference where the notice appears again when we remove the notice first and add another one. I recorded my screen to highlight what I mean.

without `removeNotice` (the notice just stays there):

![singlenotes](https://user-images.githubusercontent.com/9202899/97306808-409e2d80-185f-11eb-90fd-e1b4216aea36.gif)

with `removeNotice` (you can see a transition between actions):

![doublenotes](https://user-images.githubusercontent.com/9202899/97306684-177d9d00-185f-11eb-92e7-7911303437f0.gif)

I don't have a personal preference and I'm fine with both, just wanted to mention it as I copy-pasted this from the comments data layer handler.

Reply:
Thanks for clarifying 👍 

TBH, the subtle blink can be easy to miss if you're not looking at the notice right now. I'm not sure it really helps, TBH. I think I'd still prefer the simpler approach where we don't need to remove this notice. 

I'm thinking that if we consider this an issue, and we want to fix it, we'd want to fix it for all instances, not just for this one. 

That being said, if you ask me, feel free to remove those `removeNotice` instances.

Reply:
I gave this more thought. Would you be good with keeping the `removeNotice` action if we move that to the request handler (instead of the failure handler) where it would also make more sense semantically?

I looked up how other data layer handlers are doing it and it seems most of them remove the notice in case a new request (e.g. for deletion) is done and add it back in case the request fails again.

More importantly, this way we circumvent a subtle “bug” if a user hits e.g. the delete button a second time and then the request is successful (the notice won’t be removed in that case) at the moment.

Reply:
Makes sense to me, and I agree it's more semantic! Let's do that 👍  Thanks!

Reply:
done via 64e8549